File: typescript/medium/maximize-happiness-of-selected-children.ts
function maximumHappinessSum(happiness: number[], k: number): number {
    happiness.sort((a, b) => b - a); // descending sort

    let happinessSum = 0;
    let decrement = 0;

    for (let i = 0; i < k; i++) {
        const currentHappiness = Math.max(happiness[i] - decrement, 0);
        happinessSum += currentHappiness;
        decrement++;
    }

    return happinessSum;
}

File: csharp/medium/maximize-happiness-of-selected-children.cs
int MaximumHappinessSum(int[] happiness, int k)
{
    Array.Sort(happiness);
    Array.Reverse(happiness); // descending sort

    int happinessSum = 0;
    int decrement = 0;

    for (int i = 0; i < k; i++)
    {
        int currentHappiness = Math.Max(happiness[i] - decrement, 0);
        happinessSum += currentHappiness;
        decrement++;
    }

    return happinessSum;
}